src/main/java/com/adobe/testing/s3mock/FileStoreController.java

```diff
@@ -249,22 +254,34 @@ public ListBucketResult listObjectsInsideBucket(@PathVariable final String bucke
       final HttpServletRequest request) {
     final String filename = filenameFrom(bucketName, request);
     try (ServletInputStream inputStream = request.getInputStream()) {
+      final Map<String, String> userMetadata = getUserMetadata(request);
       final S3Object s3Object = fileStore.putS3Object(bucketName,
           filename,
           request.getContentType(),
           inputStream,
-          isV4SigningEnabled(request));
+          isV4SigningEnabled(request),
+          userMetadata);
 
       final HttpHeaders responseHeaders = new HttpHeaders();
       responseHeaders.setETag("\"" + s3Object.getMd5() + "\"");
       responseHeaders.setLastModified(s3Object.getLastModified());
+      addUserMetadata(responseHeaders::add, s3Object);
       return new ResponseEntity<>(responseHeaders, HttpStatus.CREATED);
     } catch (final IOException e) {
       LOG.error("Object could not be saved!", e);
       return new ResponseEntity<>(e.getMessage(), HttpStatus.INTERNAL_SERVER_ERROR);
     }
   }
 
+  private Map<String, String> getUserMetadata(HttpServletRequest request) {
+    return Collections.list(request.getHeaderNames()).stream()
+            .filter(header -> header.startsWith("x-amz-meta-"))
```

Comment:
Let's extract a constant for `"x-amz-meta-` as it used three times here.

Reply:
Makes sense :-)

I've directly amended it + force pushed - hope this is ok in this case :-)